### PR TITLE
`Development`: Add documentation regarding traefik as reverse proxy

### PR DESCRIPTION
**Pull request by dfuchss**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

We use Traefik as reverse proxy. Thus, we can provide a running configuration for the documentation :)

See: https://artemis-platform--8736.org.readthedocs.build/en/8736/dev/setup/docker-compose.html#using-traefik-as-a-reverse-proxy

### Steps for Testing
N/A

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### CODE REVIEW

**Review by sarpsahinalp — CHANGES_REQUESTED** [review]
A suggestion and a small typo

File: docs/dev/setup/docker-compose.rst [hunk 1]
```diff
@@ -164,3 +164,157 @@ Other useful commands
 - Restart a service: ``docker compose restart <name of the service>``
 - Remove all local Docker containers: ``docker container rm $(docker ps -a -q)``
 - Remove all local Artemis Docker images: ``docker rmi $(docker images --filter=reference="ghcr.io/ls1intum/artemis:*" -q)``
+
+Using Traefik as a reverse proxy
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you want to use Traefik as a reverse proxy for your Artemis instance, you can configure your setup using a single traefik.toml file and a docker-compose.yml file.
+The traefik.toml file should be located in the same directory as the docker-compose.yml file.
+You should also ensure that a acme.json file exists within your directory (permissions 600 for root).
```
Comment (sarpsahinalp) [inline docs/dev/setup/docker-compose.rst:173]:
```suggestion
You should also ensure that an acme.json file exists within your directory (permissions 600 for root).
```

**Review by sarpsahinalp — CHANGES_REQUESTED** [review]
A small typo

File: docs/dev/setup/docker-compose.rst [hunk 2]
```diff
@@ -164,3 +164,157 @@ Other useful commands
...
+The acme.json file is used to store the certificates generated by Traefik.
+
+
+The traefik.toml file should look like this:
```
Comment (sarpsahinalp) [inline docs/dev/setup/docker-compose.rst:177]:
```suggestion
The traefik.toml file should be configured as follows:
```

**Review by sarpsahinalp — CHANGES_REQUESTED** [review]
A suggestion here

File: docs/dev/setup/docker-compose.rst [hunk 3]
```diff
@@ -164,3 +164,157 @@ Other useful commands
...
+Using Traefik as a reverse proxy
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you want to use Traefik as a reverse proxy for your Artemis instance, you can configure your setup using a single traefik.toml file and a docker-compose.yml file.
+The traefik.toml file should be located in the same directory as the docker-compose.yml file.
+You should also ensure that an acme.json file exists within your directory (permissions 600 for root).
+The acme.json file is used to store the certificates generated by Traefik.
+
+
+The traefik.toml file should be configured as follows:
```
Comment (b-fein) [inline docs/dev/setup/docker-compose.rst:177]:
```suggestion
If you want to use `Traefik <https://doc.traefik.io/traefik/>`_ as a reverse proxy for your Artemis instance, you can configure your setup using a single ``traefik.toml`` file and a ``docker-compose.yml`` file.
The ``traefik.toml file`` should be located in the same directory as the ``docker-compose.yml`` file.
You should also ensure that an ``acme.json`` file exists within your directory (permissions 600 for root).
The ``acme.json`` file is used to store the certificates generated by Traefik.


The ``traefik.toml`` file should be configured as follows:
```

File: docs/dev/setup/docker-compose.rst [hunk 4]
```diff
@@ -164,3 +164,157 @@ Other useful commands
...
+      endpoint = "unix:///var/run/docker.sock"
+      exposedByDefault = false
+      network = "artemis-net" # name of the network in the docker-compose.yml file
+
+    [accessLog]
+    [accesslog.fields.names]
+      StartUTC = "drop"
+
+
+The docker-compose.yml file could look like this for an Artemis, Jenkins and MySQL setup with Traefik as a reverse proxy:
```
Comment (b-fein) [inline docs/dev/setup/docker-compose.rst:222]:
```suggestion
The ``docker-compose.yml`` file could look like this for an Artemis, Jenkins and MySQL setup with Traefik as a reverse proxy:
```

File: docs/dev/setup/docker-compose.rst [hunk 5]
```diff
@@ -164,3 +164,157 @@ Other useful commands
...
+          - "/var/run/docker.sock:/var/run/docker.sock:ro"
+        networks:
+          - artemis-net
+        environment:
+          - TZ=Europe/Berlin
+        labels:
+          - "traefik.enable=true"
+          - "traefik.http.routers.api.rule=Host(`${ARTEMIS_HOST_NAME}`) && (PathPrefix(`/api/`) || PathPrefix(`/dashboard/`)) || Path(`/dashboard`) || Path(`/api`)"
+          - "traefik.http.middlewares.api-redirect.redirectregex.regex=^https?://([^/]+)(/[^/]+)$"
+          - "traefik.http.middlewares.api-redirect.redirectregex.replacement=https://$$1$$2/"
```
Comment (b-fein) [inline docs/dev/setup/docker-compose.rst:246]: For this part of the config I’m not sure what it is doing, maybe a comment with a short explanation would be good.

**Review by b-fein — DISMISSED** [review]
I’ve not used Traefik in production before, but the config seems sensible to me from what I understand. Some formatting suggestions/nitpicks.

File: docs/dev/setup/docker-compose.rst (hunk 5 above)
Comment (dfuchss) [reply docs/dev/setup/docker-compose.rst:246]: I'll explain that part tomorrow

**Review by b-fein — APPROVED** [review]

**Review by egekurt123 — APPROVED** [review]
Added Documentation LGTM

**Review by marlon-luca-bu — APPROVED** [review]
Docs look good to me

**Review by az108 — APPROVED** [review]
LGTM

**Review by krusche — APPROVED** [review]

**Pull request merged**